src/Tasks/PopulateSupportedArchitectures.cs

```diff
@@ -0,0 +1,198 @@
+                switch (validationResult)
+                {
+                    case ManifestValidationResult.Success:
+                        PopulateSupportedArchitecturesElement(document, xmlNamespaceManager);
+                        SaveManifest(document);
+                        return true;
+                    case ManifestValidationResult.SupportedArchitecturesExists:
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+
+            return false;
+        }
+
+        private XmlDocument LoadManifest(string path)
+        {
+            XmlDocument document = new XmlDocument();
+            using (FileStream fs = File.OpenRead(path))
+            using (XmlReader xr = XmlReader.Create(fs, new XmlReaderSettings { DtdProcessing = DtdProcessing.Ignore, CloseInput = true }))
+            {
+                document.Load(xr);
+            }
+
+            return document;
+        }
+
+        private void SaveManifest(XmlDocument document)
+        {
+            ManifestPath = Path.Combine(OutputPath, Path.GetFileName(PathToManifest));
+
+            using (XmlWriter xmlWriter = XmlWriter.Create(ManifestPath, new XmlWriterSettings { Indent = true, Encoding = Encoding.UTF8 }))
+            {
+                document.Save(xmlWriter);
+            }
+        }
+
+        private ManifestValidationResult ValidateManifest(XmlDocument document, XmlNamespaceManager xmlNamespaceManager)
+        {
+            if (string.IsNullOrEmpty(ApplicationManifestPath))
+            {
+                return ManifestValidationResult.Success;
+            }
+
+            XmlNode? assemblyNode = document.SelectSingleNode(XPaths.assemblyElement, xmlNamespaceManager);
+            if (assemblyNode != null)
+            {
+                XmlNode? supportedArchitecturesNode = assemblyNode.SelectSingleNode($"//*[local-name()='{supportedArchitectures}']", xmlNamespaceManager);
+                if (supportedArchitecturesNode != null)
+                {
+                    if (!string.Equals(supportedArchitecturesNode.InnerText.Trim(), SupportedArchitecturesValue, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Log.LogErrorWithCodeFromResources("PopulateSupportedArchitectures.InvalidValueInSupportedArchitectures", supportedArchitecturesNode.InnerText);
+
+                        return ManifestValidationResult.Failure;
+                    }
+
+                    return ManifestValidationResult.SupportedArchitecturesExists;
+                }
+
+                return ManifestValidationResult.Success;
+            }
+
+            return ManifestValidationResult.Failure;
+        }
+
+        private string PathToManifest => string.IsNullOrEmpty(ApplicationManifestPath) || !File.Exists(ApplicationManifestPath)
+                ? ToolLocationHelper.GetPathToDotNetFrameworkFile(DefaultManifestName, TargetDotNetFrameworkVersion.Latest)
```

Comment:
This couples the output of the build to the running .NET Framework (and would fail on Core, right?). Should we embed the default manifest into this task instead?

Reply:
Added manifest to msbuild that will be a fallback mechanism if TargetDotNetFrameworkVersion.46 doesn't exists (taken from Roslyn)

Reply:
https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Compilation/Compilation.cs#L1920-L1924